subsys/mgmt/mcumgr/smp_bt.c

```diff
@@ -316,27 +316,30 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 	(void)k_work_cancel_delayable(&cpd->dwork);
 
 	/* Clear cpd. */
-	cpd->latency_state = 0;
+	cpd->state = 0;
 	cpd->conn = NULL;
 }
 
 /* BT LE connection parameters updated callback. */
 static void le_param_updated(struct bt_conn *conn, uint16_t interval,
 			     uint16_t latency, uint16_t timeout)
 {
+	const struct bt_le_conn_param *smp_params = SMP_BT_CONN_PARAMS;
 	struct conn_param_data *cpd = get_conn_param_data(conn);
 
-	if (latency == 0) {
-		cpd->latency_state |= CONN_LOW_LATENCY_ENABLED;
-		cpd->latency_state &= ~CONN_LOW_LATENCY_REQUIRED;
+	if (((interval >= smp_params->interval_min) && (interval <= smp_params->interval_max)) &&
+	    (latency == smp_params->latency) &&
+	    (timeout == smp_params->timeout)) {
+		cpd->state |= SMP_CONN_PARAMS_ENABLED;
+		cpd->state &= ~SMP_CONN_PARAMS_REQUIRED;
 		(void)k_work_reschedule(&cpd->dwork, K_MSEC(RESTORE_TIME));
 	} else {
-		cpd->latency_state &= ~CONN_LOW_LATENCY_ENABLED;
+		cpd->state &= ~SMP_CONN_PARAMS_ENABLED;
```

Comment:
I wonder if we should request peripheral prefered parameters here if they are not already active - to make behavior more consistent?

Reply:
In general if peer slightly modifies our SMP connection parameters when our request is handled we result in getting stuck with parameters that are close to the ones required by SMP forever.